ts/src/renderer/phaser/scenes/MobileControlsScene.ts

```diff
@@ -1,15 +1,143 @@
-class MobileControlsScene extends Phaser.Scene {
+type MobileControlKey =
+	'movementWheel' |
+	'lookWheel' |
+	'lookAndFireWheel' |
+	string;
+
+class MobileControlsScene extends PhaserScene {
+
+	controls: Phaser.GameObjects.Container;
+	joysticks: PhaserJoystick[] = [];
 
 	constructor() {
 		super({ key: 'MobileControls' });
 	}
 
-	create (): void { // TODO port mobile controls logic
-		/*this.mobileControls.zIndex = 10;
+	init (): void {
+
+		// enabling four mobile pointers
+		this.input.addPointer(3);
+
+		const controls = this.controls = this.add.container();
+		this.resize();
+		this.scale.on(Phaser.Scale.Events.RESIZE, this.resize, this);
+
+		const joysticks = this.joysticks;
+
+		ige.mobileControls.on('add-control',(
+			key: MobileControlKey,
+			x: number,
+			y: number,
+			w: number,
+			h: number,
+			settings: JoystickSettings
+		) => {
+
+			switch (key) {
+
+				case 'movementWheel':
+				case 'lookWheel':
+				case 'lookAndFireWheel':
+
+					new PhaserJoystick(this, x, y, settings);
+
+					break;
+
+				default:
+
+					const text = key.toUpperCase();
+
+					const button = this.add.image(x, y, 'mobile-button-up')
+						.setDisplaySize(w, h)
+						.setOrigin(0)
+						.setAlpha(0.6);
+					controls.add(button);
+
+					if (text === 'BUTTON1') {
+						const icon = this.add.image(
+							x + w/2, y + h/2,
+							'mobile-button-icon'
+						);
+						icon.setScale(0.5);
+						controls.add(icon);
+					} else {
+						const label = this.add.text(
+							x + w/2, y + h/2,
+							text,
+							{
+								fontFamily: 'Arial',
+								color: '#ffffff',
+								align: 'center'
+							});
+						label.setFontSize(24);
+						label.setOrigin(0.5);
+						controls.add(label);
+					}
+
+					button.setInteractive();
+
+					let clicked = false;
```

Comment:
could be my usual overthinking, but I just want to check. Is this the correct scope for this boolean?

Reply:
i can explain further. to me it looks like it exists in the event callback scope, but then its used in two internal functions (that are in the same scope, so I understand that), but does the boolean persist even after the callback ends?